MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/adapter/ExampleAdapter.java

```diff
@@ -64,10 +65,18 @@ public int getItemViewType(int position) {
 
   @Override
   public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
-    if (holder.getItemViewType() == 0) {
-      ExampleItemModel detailItem = dataSource.get(position);
-      ViewHolder viewHolder = (ViewHolder) holder;
+    if (holder.getItemViewType() != 0) {
+      return;
+    }
+
+    ExampleItemModel detailItem = dataSource.get(position);
+    ViewHolder viewHolder = (ViewHolder) holder;
+
+    if (Build.VERSION.SDK_INT < detailItem.getMinSdkVersion()) {
+      return;
+    }
 
+    if (Build.VERSION.SDK_INT >= detailItem.getMinSdkVersion()) {
```

Comment:
There is no need to do that comparison as you know already from line 75 that this is true

Reply:
Right 🤦‍♂️    Removed